check_gha_pinning/__init__.py

```diff
@@ -71,13 +96,11 @@ def _check(line: str) -> None:
 
 def check_pinning(file: pathlib.Path) -> list[str]:
     try:
-        workflow = ruamel.yaml.YAML().load(file)
+        workflow: dict = ruamel.yaml.YAML().load(file)
     except ruamel.yaml.YAMLError as ex:
         return [f"Error parsing yaml: {ex}"]
-    try:
-        jobs = workflow.get("jobs", {})
-    except AttributeError:
-        return []  # yaml is not a mapping on top level
+
+    jobs: dict = workflow.get("jobs", {})
```

Comment:
without this the pre-commit hook will crash on an empty yaml. I think returning [] (and passing) would be better than to present the user with a traceback

Reply:
Ooh. Yep. I see that now. Added a test, and will reinstate the except!